### DISCUSSION TITLE
Darstellungsfilter für Datalinq

### DISCUSSION THREAD

**Discussion by thomaskofler** [e-netze/datalinq-community#5] [General]
Wäre es möglich einen Darstellungsfilter wie im Webgis auch für den Datalinq (über SQL?) zu implementieren? Wir haben nämlich das Problem, dass wir im Webgis den Darstellungsfilter verwenden und über die Abfrage in den Datalinq einspringen und da zieht der Filter dann nicht mehr. Somit könnte man die URL z.B. Gemeindenummer abändern und man sieht auch die Daten, die man eigentlich nicht sehen sollte :)

**Comment by jugstalt** [comment]
Das ist leider schwer möglich. DataLinq Abfragen sind komplett entkoppelt vom CMS und eigentlich auch vom WebGIS Viewer. Die Abfragen müssen nicht zwingend aus dem WebGIS aus aufgerufen werden!
Eigentlich muss man sich hier eher die "Logik" der Abfrage überleben. Einige Fragen wären zu diesen Thema interessant:

* Wie wird der Darstellungsfilter gesetzt (vom Anwender oder versteckt, beispielsweise über einen Rollenparameter)?
* Wie wird die DataLinq Abfrage aus dem WebGIS aufgerufen? Aus der Ergebnistabelle? Sind dort nicht alle nötigen Parameter drin, um sie an die DataLinq Abfrage zu übergeben?

**Reply by thomaskofler** [reply]
Darstellungsfilter wird über Rollenparameter gesetzt z.B. [role-parameter: GEMNR like {0},OR]. 
Wenn man z.B. eine Fläche abfragt, dann erscheint im Ergebnisfenster ein Link, wo die Gemeindenummer übergeben wird:
../../../api5/datalinq/report/test@test@test?GEMNR=[GEMNR]

Wenn z.B. ein Amtsleiter von Kuchl Webgis öffnet, dann sieht er nur die Flächen seiner Gemeinde, wenn er dann den Link zum Dashobard aufruft, könnte er die GEMNR einfach austauschen und dann sieht er auch die Inhalte der anderen Gemeinden im Dashboard.

**Comment by jugstalt** [comment]
Ok, dann ist die Beschreibung der Anforderung irreführend gewesen.
Was du möchtest ist, dass Rollenparameter in eine DataLinq Abfrage einfließen können, zB
  
    SELECT * FROM GEM_DATEN WHERE GEM_NR={{role-parameter:gemnr}}

Richtig?

**Reply by thomaskofler** [reply]
ja genau

**Comment by gis-eni** [comment]
Funktionalität kommt in nächsten Build 5.23.4301, genau wie hoben beschreiben:

``` SQL
SELECT * FROM GEM_DATEN WHERE GEM_NR={{role-parameter:gemnr}}
```

Ich kann das bei mir leider nicht 100%ig testen, falls es nicht funktionieren sollte bitte melden.
